src/Cake.Common/Tools/DotNetCore/Publish/DotNetCorePublisher.cs

```diff
@@ -114,11 +114,19 @@ private ProcessArgumentBuilder GetArguments(string path, DotNetCorePublishSettin
             }
 
             // Self contained
-            if (settings.SelfContained)
+            if (settings.SelfContained.HasValue)
             {
-                builder.Append("--self-contained");
+                if (settings.SelfContained.Value)
+                {
+                    builder.Append("--self-contained");
+                }
+                else
+                {
+                    builder.Append("--self-contained false");
```

Comment:
Can we divide this into two separate appends instead?

Reply:
You mean two separate ifs, one for true and another if for false?

Reply:
No, like this:

```csharp
builder.Append("--self-contained");
builder.Append("false");
```

Reply:
oh... sure